### PR TITLE
[16.0][ADD] web_help

### PR DESCRIPTION
**Pull request by tarteo**
This module introduces a new way to guide the user through Odoo.
This module is created because **the standard Odoo tours:**

1. forces the user to create records;
2. cannot be reopened;
3. is only available for the admin user;
4. the bubbles are not obvious.

demo

You can see the demo live by going to the users list view (Settings > Users & Companies > Users)

### CODE REVIEW

**Comment by simahawk** [comment]
@sbidoul do we have some problems w/ Runboat? I see is not appearing on my repos/PRs

**Comment by simahawk** [comment]
> @sbidoul do we have some problems w/ Runboat? I see is not appearing on my repos/PRs

Seems to be only an issue w/ checks links on the PR since I can find the builds here https://runboat.odoo-community.org/webui/builds.html?repo=OCA/web&target_branch=16.0

**Comment by simahawk** [comment]
@tarteo nice toy here :)

Questions:

1. I don't see anything popping up on runboat. Is there any manual action required?
2. how do you track if a help tour has already been used/discarded?

**Comment by sbidoul** [comment]
@simahawk I don't see anything wrong with runboat here?

**Review by legalsylvain — COMMENTED** [review]
Hi. 
I created a new user on runboat. test@test.com // test
I logged with the new user
I go to Settings > Users & Companies > Users
-> Nothing occures.

Did i missed something ? 

tested on chromium  110.0.5481.100 and Firefox 110.0.

File: web_help/readme/USAGE.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+**Demo:**
+
+  .. image:: ../static/description/demo.gif
+
+You can see the demo live by going to the users list view (Settings > Users & Companies > Users)
+It's easy to create your own guides, please refer to ``static/src/user_trip.js``
```
Comment (legalsylvain) [inline web_help/readme/USAGE.rst]:
```suggestion
It's easy to create your own guides, please refer to ``static/src/user_trip.esm.js``
```

**Comment by tarteo** [comment]
Hi @simahawk @legalsylvain!
I forgot to mention that you'll need to click the little '?' button next to the view switcher. I'll add this to the readme next week. 
The help tours are meant to be accessible even after the user already did them. But it would be nice addition to be able to make some tours activated automatically on the first time. Thanks!
Have a nice weekend!

**Comment by legalsylvain** [comment]
Hi @tarteo.
> I forgot to mention that you'll need to click the little '?' button next to the view switcher

thanks for the explanation. works great ! 

I have however a design question. I don't think the place near the view switcher is optimal.

I think it should be better near the title of the page. In a tree view : 

image

**Reason :** Seeing all the different kind of view, the title seems to be the only place always displayed.

- in a classic form view the title (breadcrumb) is available, but the view switcher is not : 

image

- In a pop up form view the title is available, but the view switcher is not : 

image

Any reason to have chosen the view switcher place ? technical ? UX ? 

Note : I agree that it is partly a matter of taste.

Regards.

File: web_help/static/src/trip.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+<templates>
+    <t t-name="web_help.TripStep" owl="1">
+        <div>
+            <div class="pb-4">
+                <t t-raw="content" />
+            </div>
+
+            <button
+                class="btn btn-lg btn-outline-light web_help_cb_button mr8"
+            >Got it</button>
```
Comment (legalsylvain) [inline web_help/static/src/trip.xml]:
```suggestion
            >Next</button>
```

Nitpicking + Not blocking : I find that "Got it" can mean "stop the trip", while "Next" is more explicit.

File: web_help/static/src/trip.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            <button
+                class="btn btn-lg btn-outline-light web_help_close pull-right"
+            >Close</button>
```
Comment (legalsylvain) [inline web_help/static/src/trip.xml]:
```suggestion
            >Cancel</button>
```

Nitpicking + Not blocking : I find that "Close" can mean "close the current step", while "Cancel" is more explicit.

File: web_help/static/src/trip.xml (hunk 2 above)
Comment (tarteo) [reply web_help/static/src/trip.xml]: I've changed it so it's easy to change the button text or the whole template.

**Comment by tarteo** [comment]
@legalsylvain I've made some changes:
* I've added the button also to the dialog
* Made it easy to change the template or the text of the buttons
* Fixed the readme
changepassword

**Comment by legalsylvain** [comment]
Hi @tarteo. I tested again. Looks great !
For the button in the popup it is well displayed.
But for the Button in the tree view it is still near the sitcher. As a result it is not displayed in small screen (phone). Did you faced the same issue ?
I think putting the button near the title could fix this issue.

**Comment by tarteo** [comment]
@legalsylvain For me it's visible also on small screens. I think it's web_responsive hiding I'll check it out and fix it.

**Comment by tarteo** [comment]
@legalsylvain It was not web_responsive that module is not yet ready for 16.0 :smile: It works also now for small screens.

image

image

**Comment by simahawk** [comment]
> @simahawk I don't see anything wrong with runboat here?

image

@sbidoul I mean this (here from a v14 build) :)

**Review by legalsylvain — APPROVED** [review]
Thanks for the changes.

functional review : All is OK for me. (no code review).

2 points : 
 
1. I still think that having the "?" button always in the same place (near the title) could be better for end user. (but not a blocking point, as it is also a matter of taste)
2. Do you though to have a shorcut key to display the next step. I just tested on runboat, and to display the next step, I have to click with the cursor on a button that is moving each time ! I think that catching "Enter" or "Space" input could be a great improvment, specially for long tour, with a lot of step. (you can add it in the roadmap.rst, if you think this point relevant).

thanks for your contribution and your reactivity during the review.

**Comment by tarteo** [comment]
> 2\. Do you though to have a shorcut key to display the next step. I just tested on runboat, and to display the next step, I have to click with the cursor on a button that is moving each time ! I think that catching "Enter" or "Space" input could be a great improvment, specially for long tour, with a lot of step. (you can add it in the roadmap.rst, if you think this point relevant).

That sounds really nice. I've added it to the roadmap :)

**Review by MiquelRForgeFlow — APPROVED** [review]

**Pull request merged**